### PR TITLE
Simplify GridActions GUI

### PR DESCRIPTION
**Pull request by med-hedi**
Signed-off-by: Mohamed Hedi Khefacha <mohamed.hedi@rte-france.com>

### CODE REVIEW

File: src/utils/rest-api.js [hunk 1]
```diff
@@ -77,3 +77,26 @@ export function deleteListByName(name) {
...
     });
 }
+
+/**
+ * Rename list by name
+ * @param oldNameList
+ * @param newNameList
+ * @returns {Promise<Response>}
+ */
+export function renameListByName(oldNameList, newNameList) {
```
Comment (ChamseddineBhd) [inline src/utils/rest-api.js]: oldNameList -> oldName  (List is already in the function name)
newNameList -> newName

File: src/translations/en.json [hunk 2]
```diff
@@ -1,21 +1,22 @@
...
     "listName": "List name",
     "placeholderEditor": "Insert your groovy script here",
     "saveNewListTitle": "Save new list",
     "saveNewListMsg": "Add the script and save the new list before going to another",
-    "alertDeleteList": "Please select a list before deleting"
+    "alertDeleteList": "Please select a list before deleting",
+    "rename": "Rename",
+    "delete": "Delete",
+    "renameList": "Rename list",
+    "newNameList": "New name list"
```
Comment (ChamseddineBhd) [inline src/translations/en.json]: New list name ?

File: src/translations/en.json [hunk 3]
```diff
@@ -1,21 +1,22 @@
 {
-    "contingencyTitle": "Contingencies lists",
-    "contingencyListIsEmpty": "Contingency list is empty",
+    "contingencyTitle": "Contingency lists",
+    "contingencyListIsEmpty": "Contingency lists is empty",
```
Comment (ChamseddineBhd) [inline src/translations/en.json]: are empty?

File: src/components/contingency-list.js [hunk 4]
```diff
@@ -437,6 +462,10 @@ const Contingency = () => {
         });
     }, [dispatch]);
 
+    useEffect(() => {
+        getAllContngencyLists();
```
Comment (ChamseddineBhd) [inline src/components/contingency-list.js]: getAllContngencyLists-> getAllContingencyLists

File: src/components/contingency-list.js [hunk 5]
```diff
@@ -424,10 +434,25 @@ const Contingency = () => {
...
+        setRenameList(true);
+    };
+
+    /**
+     * Get all contingency lists on load page
+     **/
+    const getAllContngencyLists = useCallback(() => {
         getContingencyLists().then((data) => {
             if (data) {
                 setFirstItemInList(data[0]);
```
Comment (ChamseddineBhd) [inline src/components/contingency-list.js:480]: data could be empty so data[0] -> error !

**Review by ChamseddineBhd — COMMENTED** [review]
Other remarks : 
when i rename a list the script disappear but the save button is not disabled !

**Review by geofjamg — CHANGES_REQUESTED** [review]
Usually, when a rename dialog is open, it is filled with current name. Could you do this too?
Also there is no UI feedback if we you a name already take by another list. Button should be disabled in this case.
Also why is cancel button enabled when the groovy code has not been modified? It should only be available when changed has been made but not yet saved.

**Comment by ChamseddineBhd** [comment]
> Usually, when a rename dialog is open, it is filled with current name. Could you do this too?
> Also there is no UI feedback if we you a name already take by another list. Button should be disabled in this case.
> Also why is cancel button enabled when the groovy code has not been modified? It should only be available when changed has been made but not yet saved.

There's no feedback for list deletion also (but this is out of this PR) !

**Review by ChamseddineBhd — APPROVED** [review]

**Comment by med-hedi** [comment]
@geofjamg I have fixed all the returns, concerning the popup I prefer do it in a separate pull request

**Pull request merged**